### DISCUSSION TITLE
Problems with plotting multiple curves onto one plot.

### DISCUSSION THREAD

**Discussion by a-m-a-n-d-a** [asreview/asreview#1192] [Troubleshooting (report errors in the "Issues" tracker)] [answered]
I wanted to overlay multiple WSS curves onto one plot. However, I ran into two problems.

**Problem 1**
The visualisation code that worked for me a couple months ago wasn't able to work because a ModuleNotFound Error.

```
import matplotlib.pyplot as plt
from asreview.analysis import Analysis
from asreviewcontrib.visualization.plot import Plot

file = r'C:\Users\aa\Desktop\cluster_double.h5'
file2 = r'C:\Users\aa\Desktop\max_triple.h5'

with Plot.from_paths([file, file2]) as plot:
    my_plot = plot.new(plot_type='inclusion')
    my_plot.set_grid()
    for key in list(plot.analyses):
        my_plot.add_wss(key, 90, add_value=True, add_text=True)
        my_plot.add_rrf(key, 10, add_value=True, add_text=True)
        my_plot.add_random(add_text=False)
        my_plot.set_legend('lower right')
        my_plot.show()

my_plot.set_xlim('lowerlimit', 'upperlimit')

ModuleNotFound Error: No module named 'asreview.analysis'
```
I wasn't sure why the module could not be found when everything was working fine previously. 

**Problem 2**
I was trying to find another way to overlay multiple WSS curves. The writeup for the ASReview Insights extension (https://github.com/asreview/asreview-insights) seemed really promising. However, the file needs to be an ASReview file (.asreview). My files are h5 files from ASReview Simulation. Is there a way to convert my h5 files into asreview files in order for me to try out the ASReview Insights extension?

**Comment by PeterLombaers** [comment, marked as answer]
I think your visualization code is not working because it uses parts of ASReview version 0.x that no longer exist in version 1.0. All the visualization tools have been moved to asreview-insights. But as you already noticed, it won't be possible to use your `.h5` simulation files in ASReview version 1.0 or asreview-insights. I'm afraid there is no converter from `.h5` to the new format, there is only a converter for `.json` files, since these are used in ASReview Lab. I think there are two solutions though:

**Solution 1: Run the simulations again with ASReview v1.0**
If your simulations don't take too long to run, this might be the easiest option. You'll get the same results, only now in a `.asreview` file. Then you can use the [asreview-insight API](https://github.com/asreview/asreview-insights#example-multiple-curves-in-one-plot) to plot multiple WSS curves in one figure. Added benefit is that the `.asreview` files take less space than the `.h5` files 

**Solution 2: Create a virtual environment with ASReview v0.19**
It is possible to have multiple versions of a Python package running next to each other on a computer by using a [virtual environment](https://docs.python.org/3/tutorial/venv.html). Open a command terminal, navigate to a directory where you want to create the virtual environment (in this case, the directory where you have the code that runs your simulations probably makes sense) and run:
```
python3 -m venv asreview_v0_venv
```
This should create a folder `asreview_v0_venv` in your current directory. It basically contains a copy of your current Python installation. To activate the virtual environment:

- On Windows, run ```asreview_v0_venv/Scripts/activate.bat```
- On Unix or MacOS, run ```source asreview_v0_venv/bin/activate```

Your command line prompt should now show `(asreview_v0_venv)` at the start of the line, to indicate that you are using the virtual environment. Now you can run:

```pip install asreview==0.19```

This installs ASReview version 0.19 in the virtual environment. Any Python commands you run while the virtual environment is active will use ASReview v0.19 instead of v1. If you now run your Python visualization code, it should work just as it used to before v1. To deactivate the virtual environment, just close the command line prompt, or run `deactivate` in the command line. When the virtual environment is not active, Python commands will use ASReview v1.

**Reply by J535D165** [reply]
To add to solution 1: Simulations run a lot faster in version 1.x!
